coursier.bzl

```diff
@@ -1043,6 +1043,10 @@ def _coursier_fetch_impl(repository_ctx):
         primary_artifact_path = infer_artifact_path_from_primary_and_repos(primary_url, repository_urls)
 
         mirror_urls = [url + "/" + primary_artifact_path for url in repository_urls]
+        if primary_url in mirror_urls:
```

Comment:
It wasn't obvious to me that this is guaranteed to be the case, but I also didn't see how it couldn't be.

Reply:
Ahh, auth makes it not true.